File: LOTS/Projet/lots-java-template/Templates/src/molecules/Molecule.java
package molecules;
import java.util.Iterator;

import graphe.*;

public class Molecule extends Graphe<Molecule, Atome, Liaison> {

	public Molecule(){
		super();
		this.nomGraphe = "Molecule" + num;
	}

	public void afficherGraphe() {
		System.out.println("Molecule : " + this.nomGraphe);
		//Affichage des atomes
		System.out.print("Liste des atomes : " );
		Iterator<Atome> i = ListeSommets.iterator();
		while (i.hasNext()){
			Atome somm=i.next();
			System.out.print(" " + somm.getNom());
		}
		System.out.println("");
		//Affichage des liaisons
		System.out.print("Listes des liaisons : ");
		Iterator<Liaison> j = ListeAretes.iterator();
		while (j.hasNext()){
			Liaison a=j.next();
			System.out.print(" " + a.getNom() + " ( " + a.getNomSommetsLies() + ")");
		}
		
		System.out.println("\n");
	}

}

File: LOTS/Projet/scala/Scala-lots/src/molecule/Molecule.scala
package molecule
import virtuel.Arete;
import virtuel.Graphe;
import virtuel.Sommet;

object numG{
  var current = 0
  def inc = {current += 1; current}
}

class Molecule extends Graphe {
  
  override type virtualGraphe = Molecule
  override type virtualSommet = Atome
  override type virtualArete = Liaison
  
  
    nomGraphe = "Molecule" + numG.inc;
  
  
  
  override def afficherGraphe(){
    System.out.println("Molecule : " + this.nomGraphe);
    //Affichage des sommets
    System.out.print("Liste des atomes : " )
    val it = listeSommets.iterator
    while(it.hasNext() ) {
     System.out.print(" " + it.next().getNom)
    }
    System.out.println("");
    //Affichage des aretes
    System.out.print("Listes des liaisons : ");
    val j = listeAretes.iterator();
    while (j.hasNext()){
      
      var ar : virtualArete = j.next()
      
      System.out.print(" " + ar.getNom + " " + ar.getNomSommetsLies);

    }
    
    System.out.println("\n");
    
  }
  
}